util.lua

```diff
@@ -14,4 +14,39 @@ function M.join(parts)
 	return str
 end
 
+function M.localPath()
+	local str = debug.getinfo(2, 'S').source:sub(2)
+	return str:match '(.*[/\\])'
+end
+
+-- check if file exists
+function M.exists(path)
+	local f = system.get_file_info(path)
+	if f ~= nil then return f.type == 'file' else return false end
+end
+
+-- check if path is a directory or not
+function M.isDir(path)
+	local f = system.get_file_info(path)
+	if f ~= nil then return f.type == 'dir' else return false end
+end
+
+-- execute proecess
+function M.exec(cmd, opts)
+  local proc = process.start(cmd, opts or {})
+  if proc then
+    while proc:running() do
+      coroutine.yield(0.1)
+    end
+    return (proc:read_stdout() or '<no stdout>\n') .. (proc:read_stderr() or '<no stderr>'), proc:returncode()
+  end
+  return nil
+end
+
+-- remove directory
+function M.rmDir(path)
+  M.exec(PLATFORM == 'Windows' and
+    { 'cmd', '/c', 'rmdir ' .. path } or
+    { 'sh', '-c', 'rm -rf ' .. path })
+end
```

Comment:
isn't there a function for this in lite xl core?

Reply:
There is  the following function:
 - `system.rmdir(dir)` removes a directory. Use this instead of `os.remove().` but **The directory must be empty.**

As the directory will not be empty I wrote this stupid (and ugly) function...

Reply:
1. This is highly unsafe. You didn't even attempt to quote the parameters.
2. You don't need to start a shell for this. Just `rm` is sufficient.

Reply:
I do agree, I did copy some snippet from somewhere to quickly develop a functional tool, I rewrote using lite-xl system functions